runtime/j9vm/java11vmi.c

```diff
@@ -142,43 +142,55 @@ hashPackageTableDelete(J9VMThread * currentThread, J9ClassLoader * classLoader,
 	return rc;
 }
 
-/** @todo The strings below need to be NLS and also provide more debug info. */
+/**
+ * A modularity helper method to throw an exception according to the incoming error code.
+ *
+ * @param[in] currentThread the current J9VMThread
+ * @param[in] errCode a modularity error code
+ *
+ * @return Void
+ */
 static void
 throwExceptionHelper(J9VMThread *currentThread, UDATA errCode)
 {
 	if (ERRCODE_SUCCESS != errCode) {
-		J9JavaVM const * const vm = currentThread->javaVM;
-		J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+		OMRPORT_ACCESS_FROM_J9VMTHREAD(currentThread);
+		U_32 moduleName = 0;
+		U_32 messageNumber = 0;
 		const char *msg = NULL;
 
 		switch (errCode) {
 		case ERRCODE_GENERAL_FAILURE:
-			msg = "general failure";
+			moduleName = J9NLS_VM_MODULARITY_GENERAL_FAILURE__MODULE;
+			messageNumber = J9NLS_VM_MODULARITY_GENERAL_FAILURE__ID;
 			break;
 		case ERRCODE_PACKAGE_ALREADY_DEFINED:
-			msg = "a package in the list has already been defined";
+			moduleName = J9NLS_VM_MODULARITY_PACKAGE_ALREADY_DEFINED__MODULE;
+			messageNumber = J9NLS_VM_MODULARITY_PACKAGE_ALREADY_DEFINED__ID;
 			break;
 		case ERRCODE_MODULE_ALREADY_DEFINED:
-			msg = "the module has already been defined";
+			moduleName = J9NLS_VM_MODULARITY_MODULE_ALREADY_DEFINED__MODULE;
+			messageNumber = J9NLS_VM_MODULARITY_MODULE_ALREADY_DEFINED__ID;
 			break;
 		case ERRCODE_HASHTABLE_OPERATION_FAILED:
-			msg = "hash operation failed";
+			moduleName = J9NLS_VM_MODULARITY_HASH_OPERATION_FAILED__MODULE;
+			messageNumber = J9NLS_VM_MODULARITY_HASH_OPERATION_FAILED__ID;
 			break;
 		case ERRCODE_DUPLICATE_PACKAGE_IN_LIST:
-			msg = "the list contains duplicate packages";
+			moduleName = J9NLS_VM_MODULARITY_DUPLICATED_PACKAGE_FOUND__MODULE;
+			messageNumber = J9NLS_VM_MODULARITY_DUPLICATED_PACKAGE_FOUND__ID;
 			break;
 		case ERRCODE_MODULE_WASNT_FOUND:
-			msg = "module was not found";
+			moduleName = J9NLS_VM_MODULARITY_MODULE_NOT_FOUND__MODULE;
+			messageNumber = J9NLS_VM_MODULARITY_MODULE_NOT_FOUND__ID;
 			break;
 		case ERRCODE_PACKAGE_WASNT_FOUND:
-			msg = "package was not found";
-			break;
-		default:
```

Comment:
The `default` case should be retained; using `NLS Failure` instead is not reasonable behavior.

Reply:
Opened 
* https://github.com/eclipse-openj9/openj9/pull/19019